libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -338,12 +390,230 @@ def _search_store(
+            # an AND on the top if nothing is specified
+            if len(value) != 1:
+                raise ValueError(
+                    "Invalid filter condition. Expected a value which "
+                    "is a dictionary with a single key that corresponds to an operator "
+                    f"but got a dictionary with {len(value)} keys. The first few "
+                    f"keys are: {list(value.keys())[:3]}"
+                )
+            operator, filter_value = list(value.items())[0]
+            # Verify that operator is an operator
+            if operator not in SUPPORTED_OPERATORS:
+                raise ValueError(
+                    f"Invalid operator: {operator}. "
+                    f"Expected one of {SUPPORTED_OPERATORS}"
+                )
+        else:  # Then we assume an equality operator
+            operator = "$eq"
+            filter_value = value
+
+        if operator in COMPARISONS_TO_NATIVE:
+            operation = COMPARISONS_TO_NATIVE[operator]
+            native_result = func.JSON_VALUE(
+                self._embedding_store.content_metadata, f"$.{field}"
+            )
+            native_operation_result = operation(native_result, str(filter_value))
+            return native_operation_result
+
+        elif operator in NUMERIC_OPERATORS:
+            operation = NUMERIC_OPERATORS[str(operator)]
+            numeric_result = func.JSON_VALUE(
+                self._embedding_store.content_metadata, f"$.{field}"
+            )
+            numeric_operation_result = operation(numeric_result, filter_value)
+
+            if not isinstance(filter_value, str):
+                numeric_operation_result = operation(
+                    cast(numeric_result, Numeric(10, 2)), filter_value
+                )
+
+            return numeric_operation_result
+
+        elif operator == "$between":
+            # Use AND with two comparisons
+            low, high = filter_value
+
+            # Assuming lower_bound_value is a ColumnElement
+            lower_bound_value = func.JSON_VALUE(
+                self._embedding_store.content_metadata, f"$.{field}"
+            )
+
+            greater_operation = NUMERIC_OPERATORS["$gte"]
+            lesser_operation = NUMERIC_OPERATORS["$lte"]
+
+            lower_bound = greater_operation(lower_bound_value, low)
+            upper_bound = lesser_operation(lower_bound_value, high)
+
+            # Conditionally cast if filter_value is not a string
+            if not isinstance(filter_value, str):
+                lower_bound = greater_operation(
+                    cast(lower_bound_value, Numeric(10, 2)), low
+                )
+                upper_bound = lesser_operation(
+                    cast(lower_bound_value, Numeric(10, 2)), high
+                )
+
+            return sqlalchemy.and_(lower_bound, upper_bound)
+
+        elif operator in {"$in", "$nin", "$like", "$ilike"}:
```

Comment:
do we have a constant for these operators?

Reply:
cant we create the constant in such a way that we can use it here:

So, STRING_OPERATORS = {"$in", "$nin", "$like" } ?

I dont see any usage of SPECIAL_CASED_OPERATORS

or TEXT_OPERATORS anyway.. so why not define constants in such a way that we can use them?

Reply:
Fixed